src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTask.groovy

```diff
@@ -165,27 +167,36 @@ abstract class AbstractDockerRemoteApiTask extends DefaultTask {
 
         DefaultDockerClientConfig dockerClientConfig = dockerClientConfigBuilder.build()
 
-        DockerClient dockerClient = DockerClientBuilder
-            .getInstance(dockerClientConfig)
-            .withDockerCmdExecFactory(getDockerCmdExecFactory())
-            .build()
-
+        DockerClient dockerClient = buildDockerClient(dockerClientConfig)
         // register buildFinished-hook to close docker client.
         project.gradle.buildFinished {
             dockerClient.close()
         }
         dockerClient
     }
 
-    private DockerCmdExecFactory getDockerCmdExecFactory() {
+    private DockerClient buildDockerClient(DockerClientConfig config) {
         boolean useNettyExecFactory = Boolean.valueOf(getProject().findProperty('gdpNettyExecFactory')?.toString())
             ?: Boolean.valueOf(System.getProperty('gdp.netty.exec.factory'))
             ?: Boolean.valueOf(System.getenv('GDP_NETTY_EXEC_FACTORY'))
 
         if (useNettyExecFactory) {
             logger.debug("Using " + NettyDockerCmdExecFactory.class.simpleName + " as driver for " + DockerClient.class.simpleName)
+			DockerClientBuilder
+			.getInstance(config)
+			.withDockerCmdExecFactory(new NettyDockerCmdExecFactory())
+			.build()
         }
-        return useNettyExecFactory ? new NettyDockerCmdExecFactory() : null
+        else {
+			DockerClientBuilder
+			.getInstance(config)
+			.withDockerHttpClient(new OkDockerHttpClient.Builder()
+                            .dockerHost(config.getDockerHost())
+                            .sslConfig(config.getSSLConfig())
+							.connectTimeout(30 * 100)
```

Comment:
Was there a reason why you provided a different connect timeout here than whatever is default?

Reply:
I wasn't too sure what the timeout should be: -
  netty's timeout seems to default to 30 seconds
  jersey's timeout seems to default to 0
  okhttp's default timeout is 10 seconds

using the default 10 seems fine, unless you want to be consistent between netty and okhttp (in which case it should probably be (30 * 1000), or want to keep the same behaviour as jersey and set to 0.

Reply:
@bmuschko good question.

@astudentofblake I would recommend the same rule as with JVM's flags. "Don't touch them if you don't need/understand them" :D So, unless you have a good motivation to change the defaults, I would recommend keeping them.